src/machine/machine_nrf52xxx.go

```diff
@@ -14,19 +14,17 @@ func CPUFrequency() uint32 {
 
 // InitADC initializes the registers needed for ADC.
 func InitADC() {
-	return // no specific setup on nrf52 machine.
-}
-
-// Configure configures an ADC pin to be able to read analog data.
-func (a ADC) Configure(config ADCConfig) {
 	// Enable ADC.
-	// The ADC does not consume a noticeable amount of current simply by being
-	// enabled.
+	// The ADC does not consume a noticeable amount of current by being enabled.
 	nrf.SAADC.ENABLE.Set(nrf.SAADC_ENABLE_ENABLE_Enabled << nrf.SAADC_ENABLE_ENABLE_Pos)
+}
```

Comment:
So now we going to have ADC automatically enabled on NRF chips. If the comment is true, then probably it's fine.
How can we confirm that? NRF chip is particularly good for projects there extreme low power consumption is required (hibernation). Will this change jeopardise this?

Reply:
Well I would expect initialization code to be in a function called `InitADC()`, and not in `Configure()`. If(!) `InitADC()` is always called (maybe by tinygo somewhere), then yes, then the ADC will be always on. Further I do not know if the comment is correct 🤷🏻 probably @aykevl known.

If not it will only be turned on if one does `InitADC()`. I guess it will break every code that does not `InitADC()`.

This poses the question: Whats valued more coherent code or not breaking existing one?

I am defiantly more leaning to the first, but this is not my party 😄

Reply:
Right, I did assume wrong, `InitADC()` is not called in runtime init method on NRF chips, like `InitSerial()` does, for example. See https://github.com/tinygo-org/tinygo/blob/release/src/runtime/runtime_nrf.go#L29-L33
And this example confirms that, client code need to call it explicitly: https://github.com/tinygo-org/tinygo/blob/release/src/examples/adc/adc.go#L12

Not sure at the moment if this is by design or simply inconsistency that we want to fix.
@aykevl

Reply:
This is by design. It you are not using ADC then you don't have to bring it with you.

Reply:
Hm, why do we init serial unconditionally then?

Reply:
Several reasons, one of which is that without serial, you cannot use the auto-detect of 1200 baud for kicking into bootloader mode without serial.

Reply:
So leave it in? @deadprogram